scalameta/parsers/js/src/main/scala/scala/meta/parsers/JSFacade.scala

```diff
@@ -0,0 +1,80 @@
+package scala.meta
+package parsers
+
+import scala.scalajs.js
+import js.JSConverters._
+import js.annotation._
+
+object JSFacade {
+
+  // https://stackoverflow.com/a/36573183/846273
+  private[this] def mergeJSObjects(objs: js.Dynamic*): js.Dynamic = {
+    val result = js.Dictionary.empty[Any]
+    for (source <- objs) {
+      for ((key, value) <- source.asInstanceOf[js.Dictionary[Any]])
+        result(key) = value
+    }
+    result.asInstanceOf[js.Dynamic]
+  }
+
+  // https://github.com/scala-js/scala-js/issues/2170#issuecomment-176795604
+  @js.native
+  private[this] sealed trait JSLong extends js.Any
+  implicit private[this] class LongJSOps(val x: Long) extends AnyVal {
+    def toJSLong: JSLong = {
+      if (x >= 0) (x & ((1L << 53) - 1)).toDouble
+      else -((-x) & ((1L << 53) - 1)).toDouble
+    }.asInstanceOf[JSLong]
+  }
+
+  private[this] def toNode(t: Tree): js.Dynamic = {
+    val base = js.Dynamic.literal(
+      "type" -> t.productPrefix,
+      "children" -> t.children.map(toNode).toJSArray,
+      "pos" -> js.Dynamic.literal(
+        "start" -> t.pos.start.offset,
+        "end" -> t.pos.end.offset
+      )
+    )
+
+    def v[A](a: A): js.Dynamic =
+      js.Dynamic.literal("value" -> a.asInstanceOf[js.Any])
+
+    // FIXME(gabro)
+    // This is to make the facade immediately useful by showing the trees' values
+    // Ideally `.toMap` on Tree will take care of adding the '.value' attribute.
+    // See https://github.com/scalameta/scalameta/issues/948
+    val value = t match {
+      case t: Lit.Int => v(t.value)
```

Comment:
This was actually only necessary for Lit.Float and Lit.Double, the other ones can use `.value.toString`. However, I just realized we can drop all of these matches and use `.syntax` instead.

```scala
val value = t match { case _: Name | _: Lit => v(t.syntax); case _ => ...}
```

That should do the right thing for Lit.Float and Lit.Double.